homeassistant/components/asuswrt/__init__.py

```diff
@@ -1,120 +1,141 @@
 """Support for ASUSWRT devices."""
+import asyncio
 import logging
 
-from aioasuswrt.asuswrt import AsusWrt
 import voluptuous as vol
 
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
 from homeassistant.const import (
     CONF_HOST,
     CONF_MODE,
     CONF_PASSWORD,
     CONF_PORT,
     CONF_PROTOCOL,
     CONF_USERNAME,
+    EVENT_HOMEASSISTANT_STOP,
 )
+from homeassistant.exceptions import ConfigEntryNotReady
 from homeassistant.helpers import config_validation as cv
-from homeassistant.helpers.discovery import async_load_platform
-from homeassistant.helpers.event import async_call_later
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import (
+    CONF_DNSMASQ,
+    CONF_INTERFACE,
+    CONF_REQUIRE_IP,
+    CONF_SSH_KEY,
+    DATA_ASUSWRT,
+    DEFAULT_DNSMASQ,
+    DEFAULT_INTERFACE,
+    DEFAULT_SSH_PORT,
+    DOMAIN,
+    SENSOR_TYPES,
+)
+from .router import get_api
 
-_LOGGER = logging.getLogger(__name__)
+PLATFORMS = ["device_tracker", "sensor"]
 
-CONF_DNSMASQ = "dnsmasq"
-CONF_INTERFACE = "interface"
 CONF_PUB_KEY = "pub_key"
-CONF_REQUIRE_IP = "require_ip"
 CONF_SENSORS = "sensors"
-CONF_SSH_KEY = "ssh_key"
-
-DOMAIN = "asuswrt"
-DATA_ASUSWRT = DOMAIN
-
-DEFAULT_SSH_PORT = 22
-DEFAULT_INTERFACE = "eth0"
-DEFAULT_DNSMASQ = "/var/lib/misc"
-
-FIRST_RETRY_TIME = 60
-MAX_RETRY_TIME = 900
-
 SECRET_GROUP = "Password or SSH Key"
-SENSOR_TYPES = ["devices", "upload_speed", "download_speed", "download", "upload"]
+
+_LOGGER = logging.getLogger(__name__)
 
 CONFIG_SCHEMA = vol.Schema(
-    {
-        DOMAIN: vol.Schema(
-            {
-                vol.Required(CONF_HOST): cv.string,
-                vol.Required(CONF_USERNAME): cv.string,
-                vol.Optional(CONF_PROTOCOL, default="ssh"): vol.In(["ssh", "telnet"]),
-                vol.Optional(CONF_MODE, default="router"): vol.In(["router", "ap"]),
-                vol.Optional(CONF_PORT, default=DEFAULT_SSH_PORT): cv.port,
-                vol.Optional(CONF_REQUIRE_IP, default=True): cv.boolean,
-                vol.Exclusive(CONF_PASSWORD, SECRET_GROUP): cv.string,
-                vol.Exclusive(CONF_SSH_KEY, SECRET_GROUP): cv.isfile,
-                vol.Exclusive(CONF_PUB_KEY, SECRET_GROUP): cv.isfile,
-                vol.Optional(CONF_SENSORS): vol.All(
-                    cv.ensure_list, [vol.In(SENSOR_TYPES)]
-                ),
-                vol.Optional(CONF_INTERFACE, default=DEFAULT_INTERFACE): cv.string,
-                vol.Optional(CONF_DNSMASQ, default=DEFAULT_DNSMASQ): cv.string,
-            }
-        )
-    },
+    vol.All(
+        cv.deprecated(DOMAIN, invalidation_version="0.121"),
```

Comment:
They changed the version semantics to so 0.121 would be 2021.02.0

Reply:
Seems that I have to remove in this version (see PR #44156)